File: opcall/paramvalidator/validate_test.go
package paramvalidator

import (
	"errors"
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
	"github.com/opspec-io/sdk-golang/model"
)

var _ = Describe("Validate", func() {
	objectUnderTest := New()
	Context("invoked w/ nil param", func() {
		It("should return expected error", func() {
			/* arrange */
			providedValue := &model.Data{}

			expectedErrs := []error{errors.New("Validate required")}

			/* act */
			actualErrs := objectUnderTest.Validate(providedValue, nil)
			Expect(actualErrs).To(Equal(expectedErrs))
		})
	})
})

File: sdks/js/src/opspec/interpreter/opcall/params/param/file/validate.test.ts
import objectUnderTest from './validate'

describe('validate', () => {
  describe('value falsy', () => {
    it('returns expected result', () => {
      /* arrange */
      const expectedResult = [new Error('file required')]

      /* act */
      const actualResult = objectUnderTest('')

      /* assert */
      expect(actualResult).toEqual(expectedResult)
    })
  })
  describe('value truthy', () => {
    it('returns expected result', () => {
      /* arrange/act */
      const actualResult = objectUnderTest('some file')

      /* assert */
      expect(actualResult).toEqual([])
    })
  })
})
